### ISSUE TITLE
Offline VCPKG on x64-linux doesn't work when some packages depend on dynamic library testing dependencies, and using elf fix doesn't work with --only-download

### ISSUE THREAD

**Issue by Cazadorro** [microsoft/vcpkg#30709]
**Describe the bug**
Offline VCPKG on x64-linux doesn't work when some packages depend on dynamic library testing dependencies, and using elf fix doesn't work with --only-download.  

x64-linux packages will still depend on dynamic libraries *built by VCPKG*, thus require elf fix found in x64-linux-dynamic.  However this fix doesn't work in offline mode because it initiates a download even after using --only-download triplet before hand. 

**Environment**
- OS: Linux, RH7,RH8
- Compiler: GCC 11

**To Reproduce**
1. Use ./vcpkg in manifest mode with dependencies that require spurious dynamic libraries.  I couldn't isolate the library that needed it, but there was some sort of testing library that was required, I believe indireclty by mpunits.  Due to the nature of the system tested on, I cannot give direct copy paste of anything on that system.  

Dependencies were mp-units, Boost, ZMQ, and OpenCV with CUDA in the dependencies. 
3. Use --only-download and specify x64-linux on a system connected to online. 
4.  Go to offline system, attempt to build normally using manifest mode getting offline vcpkg to work (I copy the entire vcpkg directory, since the sources are the same). 
5. try to build basic main.cpp, fails not able to find dynamic dependency, which does exist and is properly created, but due to x64-linux-dynamic elf fix not existing for x64-linux (which assumes static by default) it won't work. 

Repeat the steps above for the second part, instead adding the following line to a custom triplet 

`set(VCPKG_FIXUP_ELF_RPATH ON)`

the same as x64-linux but with that line. 

When running vcpkg install, vcpkg will attempt to access the internet to download elf patch, making it impossible to use in offline mode. 

**Expected behavior**
Either x64-linux shouldn't depend on dynamic dependencies (that aren't explicitly told to the user they need to download on the host system, which is the case for a few of the dependencies vcpkg needed in general) and the first scenario should have compiled fine, or the first scenario should have compiled fine because the elf fix was included in x64-linux. 

The elf fix should also download all necessary files during --only-download mode so that it is possible to be used during offline usage of vcpkg.

**Comment by Osyotr** [comment]
I guess `vcpkg_find_acquire_program(PATCHELF)` can be executed before `portfile.cmake` is included.
Also somewhat related: https://github.com/microsoft/vcpkg/pull/27426

**Comment by dg0yt** [comment]
> I guess `vcpkg_find_acquire_program(PATCHELF)` can be executed before `portfile.cmake` is included. Also somewhat related: #27426

The key question is if we want to generally do this `vcpkg_find_acquire_program(PATCHELF)` for every port (on Linux). I don't this is done for other tools. In some cases (CMake, ninja, ..), download is done during bootstrap (`scripts/vcpkgTools.xml` ?).

But isn't the download completed by the first script port included in the set of ports being installed with `--only-downloads`? `vcpkg-cmake` is almost everywhere.

**Comment by dg0yt** [comment]
> But isn't the download completed by the first script port included in the set of ports being installed with `--only-downloads`? `vcpkg-cmake` is almost everywhere.

By the way didn't we want to enabe `VCPKG_FIXUP_ELF_RPATH` also for the official static triplet, to fix usage of `vcpkg_check_linkage(ONLY_DYNAMIC_LIBRARY)`.

**Comment by Cazadorro** [comment]
This is still an issue, and prevents us from using VCPKG in many projects, and when it can be used, we have to *also include* a *non vcpkg* path (which usually involves defaulting many libraries to be git submodules).  Right now there appears to be no way for us to use VCPKG in an offline environment, at least consistently.

**Comment by Osyotr** [comment]
@Cazadorro you can workaround this by using overlay triplet with `VCPKG_FIXUP_ELF_RPATH` and also adding overlay port with just `vcpkg_find_acquire_program(PATCHELF)` in its port file.

**Comment by Osyotr** [comment]
No you just need to add `set(VCPKG_FIXUP_ELF_RPATH ON)` to your triplet and create a port like `cache-patchelf` with a content similar to this:
`portfile.cmake`:
```cmake
set(VCPKG_POLICY_EMPTY_PACKAGE enabled)
vcpkg_find_acquire_program(PATCHELF)
```
`vcpkg.json`:
```json
{
  "name": "cache-patchelf",
  "version": "0",
  "description": "Workaround to be able to cache patchelf in --only-download mode",
  "license": null
}
```
Then add `cache-patchelf` to your project's manifest file and the next time you run `vcpkg install --only-download` it will cache patchelf.

**Comment by Cazadorro** [comment]
@Osyotr that command does not actually grab the required file unless in --only-download is *not* enabled, I don't know how to actually get it to download  using that mode.

**Comment by Osyotr** [comment]
The issue should be fixed by #37129
